lib/puppet/util/windows/process.rb

```diff
@@ -204,6 +204,37 @@ def elevated_security?
   end
   module_function :elevated_security?
 
+  def windows_major_version
+    ver = nil
+
+    FFI::MemoryPointer.new(OSVERSIONINFO.size) do |os_version_ptr|
+      os_version = OSVERSIONINFO.new(os_version_ptr)
+      os_version[:dwOSVersionInfoSize] = OSVERSIONINFO.size
+
+      result = GetVersionExW(os_version_ptr)
+
+      if result == FFI::WIN32_FALSE
+        raise Puppet::Util::Windows::Error.new(
+          "GetVersionEx(#{os_version_ptr})")
```

Comment:
It's not super useful, aside from letting you know that it's a live Ruby object.  There's not a default `to_s` for an FFI struct, for reasons I'm sure you can understand.

```
#<Puppet::Util::Windows::Process::OSVERSIONINFO:0x0000000314dc98>
```